config/openshift/manifests/prometheus-sa.yaml

```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
```

Comment:
I am not sure I understand this `ServiceAccount`, we are binding a role to read prometheus resources to it, but then where is this serviceAccount used? I don't think the `addon-operator` should require access to the prometheus resources.

Reply:
needed prometheus sa,role and rolebinding to enable in-cluster prometheus to discover targets. comfirmed by opneshift-monitoring team [https://coreos.slack.com/archives/C0VMT03S5/p1648570104344249] and https://docs.google.com/document/d/1rCKAYTrYMESjJDyJ0KvNap05NNukmNXVVi6MShISnOw/edit https://github.com/openshift/cluster-etcd-operator/tree/master/manifests